src/Authentication/Wsse.php

```diff
@@ -40,7 +40,7 @@ public function authenticate(RequestInterface $request)
         // TODO: generate better nonce?
         $nonce = substr(md5(uniqid(uniqid().'_', true)), 0, 16);
         $created = date('c');
-        $digest = base64_encode(sha1(base64_decode($nonce).$created.$this->password, true));
+        $digest = base64_encode(hash('sha512', base64_decode($nonce).$created.$this->password, true));
```

Comment:
to fully resolve #43, there should be an optional constructor argument on this class for the algorithm that defaults to the algo we want to use.

i am not sure if we should change the default algorithm - if somebody is using this library with a server that only supports sha1 but not sha512, this would be a BC break for them. in a new major version, we can change to default to a state-of-the-art hash and let people overwrite to sha1 when needed...